docs/DafnyRef/UserGuide.md

```diff
@@ -51,129 +96,315 @@ verification failure in a file that is being checked.
+
+- What is verified
+   - `--verify-included-files` (when enabled, all included files are verified, except library files, otherwise just those files on the command-line)
+   - `--relax-definite-assignment`
+   - `--track-print-effects`
+   - `--disable-nonlinear-arithmetic`
+
+- Control of the proof engine
+   - `--verification-time-limit`
+   - `--boogie`
+   - `--boogie-filter`
+
+
+#### 25.5.1.4. `dafny translate` {#sec-dafny-translate}
+
+The `dafny translate` command translates Dafny source code to source code for another target programming language.
+The command always performs the actions of `dafny resolve` and by default does the actions of `dafny verify`.
+
+In addition to generating the target source code, `dafny` may generate build artifacts to assist in compiling the generated code.
+The specific files generated depend on the target programming language.
+More detail is given in [the section on compilation](#sec-compilation).
+
+The `dafny` tool intends that the compiled program in the target language be a semantically faithful rendering of the 
+(verified) Dafny program. However, resource and language limitations make this not always possible. For example, Dafny
+can express and reason about arrays of unbounded size, not all target programming languages can represent arrays larger than the maximum signed 32-bit integer.
+
+Various options control the translation process, in addition to all those described for [`dafny resolve`](#sec-dafny-resolve) and [`dafny verify`](#sec-dafny-verify).
+
+- General options:
+   - `--no-verify` --- turns off all attempts to verify the program
+   - `--compile-verbose` --- print information about generated files
+
+- The translation target
+   - `--target` (or `-t`) --- the target language: cs (C#), java, js (Javascript), go, py (Python), cpp (C++)
+   - `--output` (or `-o`) --- location (file or folder depending on the target) of the generated file(s)
+   - `--include-runtime` --- include the Dafny runtime as source or a library in the target language
+   - `--optimize-erasable-datatype-wrapper`
+   - `--enforce-determinism`
+
+#### 25.5.1.5. `dafny build` {#sec-dafny-build}
+
+The `dafny build` command runs `dafny translate` and then compiles the result into an executable artifact for the target platform,
+such as a `.exe` or `.dll`. or executable `.jar`, or just the source code for an interpreted language.
+As with `dafny translate`, all the previous phases are also executed, including verification (unless `--no-verify` is a command-line argument).
+
+There are no additional options for `dafny build` beyond those for `dafny translate` and the previous compiler phases.
+
+TODO: OLD TEXT: The command has options that enable being specific about the build platform and architecture.
+
+#### 25.5.1.6. `dafny run` {#sec-dafny-run}
+
+The `dafny run` command performs `dafny build` and then runs the resulting executable.
```

Comment:
I also don't look at `dafny run` like this. `dafny build` should spend time optimising the binary so that executes quickly, while `dafny run` should quickly produce an executable that is allowed to be slow(er).

Reply:
OK -- adjusted the text.